lib/measured/rails/validations.rb

```diff
@@ -6,13 +14,40 @@ def validate_each(record, attribute, measurable)
     measurable_unit = record.send("#{ attribute }_unit")
     measurable_value = record.send("#{ attribute }_value")
 
-    message = options[:message] || "is not a valid unit"
+    return unless measurable_unit.present? || measurable_value.present?
+
+    record.errors.add(attribute, message) if [measurable_unit.blank?, measurable_value.blank?].any?
```

Comment:
Due to the return on line 17 this will never fail. Present is the opposite of blank. If you are returning if something isn't present, then they are blank, and therefore this line will never execute.

Sorry for the long winded sentence, that was more for me than anything.